temporal/fx.go

```diff
@@ -481,46 +481,94 @@ func ApplyClusterMetadataConfigProvider(
 	}
 	defer clusterMetadataManager.Close()
 
-	applied, err := clusterMetadataManager.SaveClusterMetadata(
-		&persistence.SaveClusterMetadataRequest{
-			ClusterMetadata: persistencespb.ClusterMetadata{
-				HistoryShardCount: config.Persistence.NumHistoryShards,
-				ClusterName:       config.ClusterMetadata.CurrentClusterName,
-				ClusterId:         uuid.New(),
-			}})
-	if err != nil {
-		logger.Warn("Failed to save cluster metadata.", tag.Error(err))
-	}
-	if applied {
-		logger.Info("Successfully saved cluster metadata.")
-		return config.ClusterMetadata, config.Persistence, nil
-	}
+	clusterData := config.ClusterMetadata
+	for clusterName, clusterInfo := range clusterData.ClusterInformation {
+		var clusterId = ""
+		if clusterName == clusterData.CurrentClusterName {
+			// Only set current cluster Id as we don't know the remote cluster Id.
+			clusterId = uuid.New()
+		}
+		// We assume the existing remote cluster info is correct.
+		applied, err := clusterMetadataManager.SaveClusterMetadata(
+			&persistence.SaveClusterMetadataRequest{
+				ClusterMetadata: persistencespb.ClusterMetadata{
+					HistoryShardCount:        config.Persistence.NumHistoryShards,
+					ClusterName:              clusterName,
+					ClusterId:                clusterId,
```

Comment:
but we don't have the clusterId from existing config for remote clusters, right?

Reply:
Yes. We unset this field in L486. With another question, does the remote cluster Id matter ?